pkg/ingester/ingester.go

```diff
@@ -326,6 +334,15 @@ func New(cfg Config, clientConfig client.Config, store Store, limits Limits, con
 		i.SetExtractorWrapper(i.cfg.SampleExtractorWrapper)
 	}
 
+	recalculateOwnedStreams := newRecalculateOwnedStreams(i.instances, i.lifecycler.ID, i.readRing, cfg.OwnedStreamsCheckInterval, util_log.Logger)
```

Comment:
we can not pass instances to that job because it will be modified in different goroutines and we will get `panics` .... 
we need to pass the reference to `func (i *Ingester) getInstances() []*instance`, it provides a safe way to get all the instances from the ingester.

Reply:
done